### DISCUSSION TITLE
@xstate/react useMachine doesn't force a rerender on Eventless transitions

### DISCUSSION THREAD

**Discussion by pakoito** [statelyai/xstate#3245] [Q&A]
I use the payload to make players go through different stages of an Arcade mode. Each stage has a new payload for the enemies. 

What I am finding is that transitioning to an Eventless transition the state machine doesn't automatically progress to the next state, leaving me with the transient state:

```
const machine = {
 ...
  initial: 'player',
  states: {
    player: {
      entry: ['reset'],
      on: {
        PLAYER: { target: 'play' }
      }
    },
    play: {
      on: {
        WIN: { target: 'win'},
        LOSE: { target: 'defeat' }
      }
    },
    win: {
      entry: ['bumpVictories'],
      on: {
        always: [                                            <<<<<<<<<<------------- THIS TRANSITION
          { target: 'play', cond: 'isNotFinal' },
          { target: 'victory', cond: 'isFinal' }
        ]
      }
    },
 ....
  }
}
```

```
const [state, send] = useMachine(machine, { devTools: true });

switch (true) {
    case state.matches({ arcade: 'play' }): {
      return <SomeComponent
         onGameEnd={(isVictory) => {
          send(isVictory ? 'WIN' : 'LOSE');
        }}
     />;
    }
  default:
     return null; <<<<<<<<<<<<<------------ The Eventless state falls here and prevents progression to a new state
   ....
}
```

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.

**Comment by pakoito** [comment]
Same as with my other report, I believe you'd repro this better with a new test than with my convoluted repo:

https://github.com/statelyai/xstate/discussions/3244#discussioncomment-2625125
